### PR TITLE
Obj inside

### PR DESCRIPTION
**Pull request by greenblitz4590**
WORKING AND BEAUTIFUL

### CODE REVIEW

File: src/main/java/edu/greenblitz/tobyDetermined/Robot.java [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 import edu.greenblitz.tobyDetermined.subsystems.swerve.SwerveChassis;
 import edu.wpi.first.net.PortForwarder;
 import edu.wpi.first.wpilibj.TimedRobot;
 import edu.wpi.first.wpilibj.livewindow.LiveWindow;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import edu.wpi.first.wpilibj.util.Color;
 
 public class Robot extends TimedRobot {
+//	private final Color kYellowTarget = new Color(0.364f, 0.522f, 0.107f);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/Robot.java:20]: Commented code

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class IntakeObjSensor extends GBSubsystem {
+	/**
+	 * it senses the object inside (cone/cube) using the color sensor,
+	 * and displays it on the dashboard
+	 */
+	public static final ColorSensorV3 cs = new ColorSensorV3(I2C.Port.kOnboard);
+	public enum Obj {
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:18]: What is obj? Maybe use GameObject?

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		CONE,
+		CUBE,
+		NONE,
+	}
+	private final Color kYellowTarget = new Color(0.477f, 0.478f, 0.063f);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:23]: Magic numbers

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class IntakeObjSensor extends GBSubsystem {
+	/**
+	 * it senses the object inside (cone/cube) using the color sensor,
+	 * and displays it on the dashboard
+	 */
+	public static final ColorSensorV3 cs = new ColorSensorV3(I2C.Port.kOnboard);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:17]: Define the value of the sensor in the constructor

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java (hunk 3 above)
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:23]: Also should be static

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	public enum Obj {
+		CONE,
+		CUBE,
+		NONE,
+	}
+	private final Color kYellowTarget = new Color(0.477f, 0.478f, 0.063f);
+	private final Color kPurpleTarget = new Color(0.27f, 0.355f, 0.357f);
+	public final ColorMatch colorMatcher = new ColorMatch();
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:25]: Give value inside the constructor

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+	private static IntakeObjSensor instance;
+
+	public Obj curObj = null;
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:29]: Obj.NONE?

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+	private IntakeObjSensor(){
+		colorMatcher.addColorMatch(kYellowTarget);
+		colorMatcher.addColorMatch(kPurpleTarget);
+		
+		colorMatcher.setConfidenceThreshold(0.9);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:35]: Magic

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.greenblitz.utils.GBCommand;
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class IntakeObjSensor extends GBSubsystem {
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:12]: Obj is not that clear, maybe IntakeSensor or IntakeGameObjectSensor
Also, maybe just add this to the intake subsystem

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+		 * The color sensor works best when within a few inches from an object in
+		 * well lit conditions (the built-in LED is a big help here!). The farther
+		 * an object is the more light from the surroundings will bleed into the
+		 * measurements and make it difficult to accurately determine its color.
+		 */
+		Color detectedColor = cs.getColor();
+
+		ColorMatchResult match = colorMatcher.matchColor(detectedColor);
+
+//		SmartDashboard.putNumber("Red", detectedColor.red);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:59]: Commented code

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+		if (match == null){
+			curObj = Obj.NONE;
+			return;
+		}
+		if (match.color == kYellowTarget) {
+			curObj = Obj.CONE;
+		} else if (match.color == kPurpleTarget){
+			curObj = Obj.CUBE;
+		}
+//		SmartDashboard.putNumber("Confidence", match.confidence);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:76]: Commented code

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+//		SmartDashboard.putNumber("Green", detectedColor.green);
+//		SmartDashboard.putNumber("Blue", detectedColor.blue);
+//		SmartDashboard.putNumber("Proxy", cs.getProximity());
+
+		/**
+		 * Run the color match algorithm on our detected color
+		 */
+		if (match == null){
+			curObj = Obj.NONE;
+			return;
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java:69]: Use the if else like you did below

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java [hunk 12]
```diff
@@ -57,8 +57,8 @@ public static SwerveChassis getInstance() {
 	
 	@Override
 	public void periodic() {
-		updatePoseEstimation();
-		field.setRobotPose(getRobotPose());
+//		updatePoseEstimation();
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java:60]: Commented code

**Review by tnrz135 — CHANGES_REQUESTED** [review]
Also, add the .idea to gitignore

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java (hunk 5 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java]: done

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java (hunk 7 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java]: now static final

Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java]: now static final

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java (hunk 9 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java]: uncommented

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java (hunk 8 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java]: now called IntakeGameObjectSensor, i think its easy to understand now

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java (hunk 11 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/IntakeObjSensor.java]: unreturnd now

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java (hunk 12 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java:60]: do we need to update the pose estimation twice?

File: src/main/java/edu/greenblitz/tobyDetermined/Robot.java (hunk 1 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/Robot.java]: not there anymore

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java (hunk 12 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java:60]: ok we dont

**Pull request merged**